packages/core/src/exporter/download-videos.ts

```diff
@@ -0,0 +1,58 @@
+import {AssetInfo} from '../app';
+
+export async function download(assets: AssetInfo[][]): Promise<void> {
+  const videoRanges: Map<string, {start: number; end: number}> = new Map();
+
+  assets.forEach(frameAssets => {
+    frameAssets.forEach(asset => {
+      if (asset.type === 'video' && asset.decoder === 'ffmpeg') {
+        if (!videoRanges.has(asset.src)) {
+          videoRanges.set(asset.src, {
+            start: asset.currentTime,
+            end: asset.currentTime,
+          });
+        } else {
+          const range = videoRanges.get(asset.src)!;
+          range.start = Math.min(range.start, asset.currentTime);
+          range.end = Math.max(range.end, asset.currentTime);
+        }
+      }
+    });
+  });
+
+  const videoDurations = Array.from(videoRanges.entries()).map(
+    ([src, {start, end}]) => ({
+      src,
+      startTime: start,
+      endTime: end,
+    }),
+  );
+
+  videoDurations.forEach(({src, startTime, endTime}) => {
+    console.log(`downloading ${src} from ${startTime}s to ${endTime}s`);
+  });
+
+  const response = await fetch(
+    '/revideo-ffmpeg-decoder/download-video-chunks',
+    {
+      method: 'POST',
+      headers: {
+        // eslint-disable-next-line
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(videoDurations),
+    },
+  );
+
+  if (!response.ok) {
+    throw new Error(`HTTP error! status: ${response.status}`);
+  }
+
+  const result = await response.json();
+
+  if (!result.success) {
+    throw new Error(`Error downloading video chunks: ${result.error}`);
+  }
+
+  console.log('finished downloading');
```

Comment:
We should probably come up with a policy on console.logs in the future. I think it's fine to leave this in for now but it makes Revideo seem more immature than it is when it emits random unstyled logs. We could maybe look at how other libraries handle this.

Same goes for error handling too. Right now we just randomly throw here and there but never really handle anything.

Reply:
fully agree